m4/ac_path_kernel_source.m4

```diff
@@ -103,19 +104,27 @@ AC_DEFUN([AC_KERNEL_CHECKS],
                            -e s/s390x/s390/)
   save_CFLAGS="$CFLAGS"
   save_CPPFLAGS="$CPPFLAGS"
-  CFLAGS=
-  CPPFLAGS="-include $kerneldir/include/linux/kconfig.h \
-            -include $kerneldir/include/linux/compiler.h \
+  CFLAGS="$KERNEL_CHECKS_CFLAGS"
+  CPPFLAGS="-include $kernelinc/include/linux/kconfig.h \
+            -include $kernelinc/include/linux/compiler.h \
             -D__KERNEL__ \
             -DKBUILD_MODNAME=\"xpmem_configure\" \
-            -I$kerneldir/include \
-            -I$kerneldir/include/uapi \
-            -I$kerneldir/arch/$srcarch/include \
-            -I$kerneldir/arch/$srcarch/include/uapi \
-            -I$kerneldir/arch/$srcarch/include/generated \
-            -I$kerneldir/arch/$srcarch/include/generated/uapi \
+            -I$kernelinc/include \
+            -I$kernelinc/include/uapi \
+            -I$kernelinc/arch/$srcarch/include \
+            -I$kernelinc/arch/$srcarch/include/uapi \
+            -I$kernelinc/arch/$srcarch/include/generated \
+            -I$kernelinc/arch/$srcarch/include/generated/uapi \
             $CPPFLAGS"
 
+  AC_CHECK_DECL([kmalloc], [], [
+    AC_MSG_ERROR([cannot run kernel module configuration checks])], [[
+    #include <linux/slab.h>
+    #include <linux/mm.h>
+    #include <linux/sched.h>
+    #include <linux/mm_types.h>
+    #include <linux/proc_fs.h>]])
```

Comment:
why needed?
maybe no need for all these includes?

Reply:
agree, not strictly needed but since we use them later in this file, I thought it made sense to add them. Then in later checks, if build fails, it's really because of a kernel check outcome.

Reply:
ok